pkg/controller/statusmanager/status_manager.go

```diff
@@ -335,11 +372,7 @@ func (status *StatusManager) SetFromPods() {
 	status.SetNotFailing(PodDeployment)
 
 	// Finally, if all containers are deployed, mark as Available
-	if len(progressing) == 0 {
+	if len(progressing) == 0 && len(status.GetContainers()) > 0 {
```

Comment:
I don't think this is valid. Empty configuration is allowed, thus we can be ready in case somebody just creates empty NetworkAddonsConfig CR without any containers listed.

Reply:
Yep, I missed the empty spec case. Fixed.